keyboards/neodox/keymaps/neodimio/keymap.c

```diff
@@ -0,0 +1,1227 @@
+// Copyright 2022 MatteoDM <matteodalmo@gmail.com>
+// SPDX-License-Identifier: GPL-2.0-or-later
+#include QMK_KEYBOARD_H
+
+/**
+*  Each layer gets a name for readability, which is then used in the keymap matrix below.
+*  The underscores don't mean anything - you can have a layer called STUFF or any other name.
+*  Layer names don't all need to be of the same length, obviously, and you can also skip them
+*  entirely and just use numbers.
+*/
+
+#ifdef QUANTUM_PAINTER_ENABLE
+#  include <qp.h>
+#  include "imgs/logo.qgf.h"
+#  include "imgs/big_font.qff.h"
+#endif
+
+#include "quantum.h"
+#include <string.h>
+#include <ctype.h>
+#include <printf.h>
+#include "color.h"
+
+#ifdef BACKLIGHT_ENABLE
+  #include "backlight.h"
+#endif
+
+
+#define _QWERTY  0
+#define _SYMB    1
+#define _MOUSE   2
+#define _NAV     3
+#define _FUNC    4
+#define _GAMING  5
+#define _ADJUST  6
```

Comment:
We highly recommend enums over defines like this, because they're a lot easier to manage, as they auto-increment.

We've actually ran into issues with gaps in layers causing bloat, and other issues. 

```suggestion
enum custom_layers {
    _QWERTY,
    _SYMB,
    _MOUSE,
    _NAV,
    _FUNC,
    _GAMING,
    _ADJUST,
};
```